Fix validation for input fields

vsramanujan opened the pull request:
- Validation for 'empty-like' strings for community name
- Show validation message at appropriate times for Review Checklist title

![captured (15)](https://user-images.githubusercontent.com/25174717/74594993-72b98780-5062-11ea-9369-c38bae4805f4.gif)


If you'd like the Review Checklist title to be trimmed before validation as well, let me know!
(Wouldn't we want to trim the values before we save them as well?)

Fixes: #206

harigopal commented on app/javascript/courses/review/components/CoursesReview__ChecklistEditor.re:
```diff
@@ -297,7 +299,9 @@ let make =
-                                resultItem |> ReviewChecklistResult.title == ""
+                                reviewChecklistItem
+                                |> ReviewChecklistItem.title
+                                |> String.length <= 2
```
These checks should have been against `resultItem`, not `reviewChecklistItem`.

vsramanujan replied:
Oh derp! 

Sorry about that, have made the change

harigopal commented on app/javascript/courses/review/components/CoursesReview__ChecklistEditor.re:
```diff
@@ -195,7 +195,9 @@ let make =
-                      reviewChecklistItem |> ReviewChecklistItem.title == ""
+                      reviewChecklistItem
+                      |> ReviewChecklistItem.title
+                      |> String.length <= 2
```
As you'd mentioned, we should trim this before checking the length.

However, since this very same check is also performed on `resultItem` (below), it would be cleaner to extract a function here. Maybe a `validTitle: string => bool` at the top-level of this module for use in both locations?

harigopal commented:
Thank you for the PR @vsramanujan!

You're right that it would be good to trim the checklist item and result titles before saving them. I suppose that the interface can _clean-up_ right before submitting the checklist to the server and updating its own state.

I've added a small server-side check to complement your client-side validation in 815628959802bd8e8cedc4aae31e1452a2bce7d6.

vsramanujan commented:
Though I've made the changes,  I am unsure whether enforcing the checklist `title` to be > 2 characters is the right thing to do.

Especially when the default view while creating a Review checklist is this: 
<img width="774" alt="Screenshot 2020-02-18 at 12 04 22 AM" src="https://user-images.githubusercontent.com/25174717/74679063-c18e2b00-51e2-11ea-88ab-24583c0a38ec.png">

Wouldn't it be ideal to maybe make it >= 2? Especially since the validation for the `Save Checklist` button anyway is written for >=2? 
https://github.com/SVdotCO/pupilfirst/blob/4eb2c0eb58f160cf67ae4e25203d41b094f9bc00/app/javascript/courses/review/components/CoursesReview__ChecklistEditor.re#L141-L157

Please let me know which one of these (the validation for the button or for the titles with the messages) needs to be changed.

harigopal commented:
@vsramanujan That's a good point. In fact, after thinking about it for a bit, I can't really tell why we're enforcing a minimum length of _two_ on the checklist item and result titles.

All we care about is the presence of _some_ string, so a `length > 0` should be sufficient as a conservative check. I have updated the server-side validation to match in d8e71bf24a49d7113bf64e6349391fe137bc0f66.

vsramanujan commented:
> @vsramanujan That's a good point. In fact, after thinking about it for a bit, I can't really tell why we're enforcing a minimum length of _two_ on the checklist item and result titles.
> 
> All we care about is the presence of _some_ string, so a `length > 0` should be sufficient as a conservative check. I have updated the server-side validation to match in [d8e71bf](https://github.com/SVdotCO/pupilfirst/commit/d8e71bf24a49d7113bf64e6349391fe137bc0f66).

I've updated the PR to match this! 

If that were the case, there are other instances of the `title.length > 2` check as well that we might need to look at.
